src/ucp/core/ucp_request.h

```diff
@@ -206,7 +205,11 @@ struct ucp_request {
                             } header UCS_S_PACKED; /* packed to avoid 32-bit
                                                       padding */
                             uint16_t       am_id;
+                            /* API flags from @ref ucp_send_am_flags enum */
                             uint16_t       flags;
+                            /* Internal implementation flags from @ref
+                             * ucp_request_am_impl_flags enum */
+                            uint8_t        impl_flags;
```

Comment:
impl_flags sounds weird.. and seems it increases "am" struct size..
maybe reduce "flags" to uint8, and rename impl_flags to internal_flags?

Reply:
or
hdr_sent: 1;

Reply:
fine for me
In this case we don't really need the ucp_request_am_internal_flags enum